core/src/main/java/greencity/webcontroller/ManagementAchievementController.java

```diff
@@ -0,0 +1,125 @@
+import greencity.service.AchievementService;
+import greencity.service.LanguageService;
+import lombok.AllArgsConstructor;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+import static greencity.dto.genericresponse.GenericResponseDto.buildGenericResponseDto;
+
+@Controller
+@AllArgsConstructor
+@RequestMapping("/management/achievement")
+public class ManagementAchievementController {
+    private final AchievementService achievementService;
+    private final AchievementCategoryService achievementCategoryService;
+    private final LanguageService languageService;
+
+    /**
+     * Method that returns management page with all {@link AchievementVO}.
+     *
+     * @param model    Model that will be configured and returned to user.
+     * @param pageable {@link Pageable}.
+     * @return View template path {@link String}.
+     * @author Orest Mamchuk
+     */
+    @GetMapping
+    public String getAllAchievement(@RequestParam(required = false, name = "query") String query, Pageable pageable,
+        Model model) {
+        Pageable paging = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), Sort.by("id").descending());
+        PageableAdvancedDto<AchievementVO> allAchievements = query == null || query.isEmpty()
+            ? achievementService.findAll(paging)
+            : achievementService.searchAchievementBy(paging, query);
+        model.addAttribute("pageable", allAchievements);
+        model.addAttribute("categoryList", achievementCategoryService.findAll());
+        model.addAttribute("languages", languageService.getAllLanguages());
+        return "core/management_achievement";
+    }
+
+    /**
+     * Method for creating {@link AchievementVO}.
+     *
+     * @param achievementPostDto dto for {@link AchievementVO} entity.
+     * @return {@link GenericResponseDto} with of operation and errors fields.
+     * @author Orest Mamchuk
+     */
+    @PostMapping
+    @ResponseBody
+    public GenericResponseDto saveAchievement(@Valid @RequestBody AchievementPostDto achievementPostDto,
+        BindingResult bindingResult) {
+        if (!bindingResult.hasErrors()) {
+            achievementService.save(achievementPostDto);
+        }
+        return buildGenericResponseDto(bindingResult);
+    }
+
+    /**
+     * Method which deteles {@link AchievementVO} by given id.
+     *
+     * @param id of {@link AchievementVO}
+     * @return {@link ResponseEntity}
+     * @author Orest Mamchuk
+     */
+    @DeleteMapping("delete/{id}")
```

Comment:
'delete' relative path not needed. Please, remove it.

Reply:
Done